### ISSUE TITLE
KeyDB restarts with assertion: Assertion `pidT >= 0' failed

### ISSUE THREAD

**Issue by jendis** [JohnSully/KeyDB#119]
**Describe the bug**
When lowering configuration paremeter _timeout_ to e.g. 10, then KeyDB restarts with logged line:

> keydb-server: fastlock.cpp:367: void fastlock_unlock(fastlock*): Assertion `pidT >= 0' failed.

** Log Files **
> @400000005df6a491051bc31c 11817:C 15 Dec 2019 22:24:23.085 # oO0OoO0OoO0Oo KeyDB is starting oO0OoO0OoO0Oo
@400000005df6a491051bcaec 11817:C 15 Dec 2019 22:24:23.085 # KeyDB version=5.1.1, bits=64, commit=00000000, modified=0, pid=11817, just started
@400000005df6a491051c4bd4 11817:C 15 Dec 2019 22:24:23.085 # Configuration loaded
@400000005df6a4910538921c 11817:M 15 Dec 2019 22:24:23.087 * Running mode=standalone, port=6382.
@400000005df6a49105390f1c 11817:M 15 Dec 2019 22:24:23.087 # Server initialized
@400000005df6a4910539c2cc 11817:M 15 Dec 2019 22:24:23.087 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with KeyDB. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. KeyDB must be restarted after THP is disabled.
@400000005df6a491053a43b4 11817:M 15 Dec 2019 22:24:23.087 * Loading RDB produced by version 5.1.1
@400000005df6a491053a62f4 11817:M 15 Dec 2019 22:24:23.087 * RDB age 755 seconds
@400000005df6a491053a9d8c 11817:M 15 Dec 2019 22:24:23.087 * RDB memory usage when created 41.17 Mb
@400000005df6a4910b769fac 11817:M 15 Dec 2019 22:24:23.192 * DB loaded from disk: 0.105 seconds
@400000005df6a4910b76a77c 11817:M 15 Dec 2019 22:24:23.192 * Ready to accept connections
@400000005df6a4910b792c04 11817:M 15 Dec 2019 22:24:23.192   Thread 0 alive.
@400000005df6a49c361f4b14 keydb-server: fastlock.cpp:367: void fastlock_unlock(fastlock*): Assertion `pidT >= 0' failed.
@400000005df6a49c37e4866c 12578:C 15 Dec 2019 22:24:34.937 # oO0OoO0OoO0Oo KeyDB is starting oO0OoO0OoO0Oo
@400000005df6a49c37e48e3c 12578:C 15 Dec 2019 22:24:34.937 # KeyDB version=5.1.1, bits=64, commit=00000000, modified=0, pid=12578, just started
@400000005df6a49c37e5324c 12578:C 15 Dec 2019 22:24:34.937 # Configuration loaded
@400000005df6a49c3801de24 12578:M 15 Dec 2019 22:24:34.939 * Running mode=standalone, port=6382.
@400000005df6a49c38025b24 12578:M 15 Dec 2019 22:24:34.939 # Server initialized
@400000005df6a49c38030ed4 12578:M 15 Dec 2019 22:24:34.939 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with KeyDB. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. KeyDB must be restarted after THP is disabled.
@400000005df6a49c380393a4 12578:M 15 Dec 2019 22:24:34.939 * Loading RDB produced by version 5.1.1
@400000005df6a49c3803b2e4 12578:M 15 Dec 2019 22:24:34.939 * RDB age 766 seconds
@400000005df6a49c3803f54c 12578:M 15 Dec 2019 22:24:34.939 * RDB memory usage when created 41.17 Mb
@400000005df6a49d02a53154 12578:M 15 Dec 2019 22:24:35.044 * DB loaded from disk: 0.105 seconds
@400000005df6a49d02a53924 12578:M 15 Dec 2019 22:24:35.044 * Ready to accept connections
@400000005df6a49d02a7b1f4 12578:M 15 Dec 2019 22:24:35.044   Thread 0 alive.

**To Reproduce**
Installed version _dev-db/KeyDB-5.1.1_ on _5.4.3-gentoo_ supervised by daemontools and logs controlled by multilog (that's why tai64 timestamp starts each log line).
Value of _server-threads_ tested with 1 and 4 with no effect, other configuration parameters were intouched.

**Comment by JohnSully** [comment]
Can you reproduce this running keydb directly (e.g. ./keydb-server directly on the shell)?  I'm having trouble reproing.

**Comment by jendis** [comment]
I did this and I was able to successfully reproduce it, here are the steps:

1. In shell 1, I run keydb:
`host /home/user # /usr/sbin/keydb-server /etc/keydb.conf --bind localhost --port 6383 --timeout 1`
> 18651:C 02 Jan 2020 15:05:00.676 # oO0OoO0OoO0Oo KeyDB is starting oO0OoO0OoO0Oo
18651:C 02 Jan 2020 15:05:00.676 # KeyDB version=0.0.0, bits=64, commit=aea333bb, modified=1, pid=18651, just started
18651:C 02 Jan 2020 15:05:00.676 # Configuration loaded
18651:M 02 Jan 2020 15:05:00.677 * Increased maximum number of open files to 10032 (it was originally set to 2048).
                                                                     
                                                                     
                                        KeyDB 0.0.0 (aea333bb/1) 64 bit
                                                                     
                                        Running in standalone mode
                                        Port: 6383
                                        PID: 18651
                                                                     
                     Like KeyDB? Star us on GitHub!                  
                                                                     
                     https://github.com/JohnSully/KeyDB              
                                                                     

18651:M 02 Jan 2020 15:05:00.678 # Server initialized
18651:M 02 Jan 2020 15:05:00.678 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with KeyDB. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. KeyDB must be restarted after THP is disabled.
18651:M 02 Jan 2020 15:05:00.678 * Ready to accept connections
18651:M 02 Jan 2020 15:05:00.678   Thread 0 alive.
18651:M 02 Jan 2020 15:05:00.678   Thread 1 alive.

2. In shell 2, I do this:
asdas
`host /home/user # keydb-cli -p 6383`
> Message of the day:
  Have questions about KeyDB?  Join us on our gitter chat: https://gitter.im/KeyDB/community
and execute command:
`127.0.0.1:6383> "LRANGE" "abcd" "1" "-1"`
